components/odh-notebook-controller/controllers/notebook_webhook.go

```diff
@@ -268,137 +264,123 @@ func (w *NotebookWebhook) Handle(ctx context.Context, req admission.Request) adm
 	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledNotebook)
 }
 
-func (w *NotebookWebhook) ClusterWideProxyIsEnabled() bool {
-	proxyResourceList := &configv1.ProxyList{}
-	err := w.Client.List(context.TODO(), proxyResourceList)
-	if err != nil {
-		return false
-	}
-
-	for _, proxy := range proxyResourceList.Items {
-		if proxy.Name == "cluster" {
-			if proxy.Status.HTTPProxy != "" && proxy.Status.HTTPSProxy != "" &&
-				proxy.Status.NoProxy != "" {
-				// Update Proxy Env variables map
-				proxyEnvVars["HTTP_PROXY"] = proxy.Status.HTTPProxy
-				proxyEnvVars["HTTPS_PROXY"] = proxy.Status.HTTPSProxy
-				proxyEnvVars["NO_PROXY"] = proxy.Status.NoProxy
-				if proxy.Spec.TrustedCA.Name != "" {
-					proxyEnvVars["PIP_CERT"] = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
-					proxyEnvVars["REQUESTS_CA_BUNDLE"] = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
-					proxyEnvVars["SSL_CERT_FILE"] = "/etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem"
-				}
-				return true
-			}
-		}
-	}
-	return false
-
-}
-
 // InjectDecoder injects the decoder.
 func (w *NotebookWebhook) InjectDecoder(d *admission.Decoder) error {
 	w.Decoder = d
 	return nil
 }
 
-func InjectProxyConfig(notebook *nbv1.Notebook) error {
-	notebookContainers := &notebook.Spec.Template.Spec.Containers
-	var imgContainer corev1.Container
+func CheckAndMountCACertBundle(ctx context.Context, cli client.Client, notebook *nbv1.Notebook) error {
 
-	// Add trusted-ca volume
-	notebookVolumes := &notebook.Spec.Template.Spec.Volumes
-	certVolumeExists := false
-	certVolume := corev1.Volume{
-		Name: "trusted-ca",
-		VolumeSource: corev1.VolumeSource{
-			ConfigMap: &corev1.ConfigMapVolumeSource{
-				LocalObjectReference: corev1.LocalObjectReference{
-					Name: "trusted-ca",
-				},
-				Items: []corev1.KeyToPath{{
-					Key:  "ca-bundle.crt",
-					Path: "tls-ca-bundle.pem",
-				},
+	configMapName := "odh-trusted-ca-bundle"
+
+	// get configmap based on its name and the namespace
+	configMap := &corev1.ConfigMap{}
+	if err := cli.Get(ctx, client.ObjectKey{Namespace: notebook.Namespace, Name: configMapName}, configMap); err != nil {
+		return err
+	}
+
+	// Search for the odh-trusted-ca-bundle ConfigMap
+	fmt.Printf("ConfigMap found on the given Namespace")
+	cm := configMap
+
+	// Validate certificates before mounting
+	if err := certValidator(cm); err != nil {
+		return err
+	}
+
+	if cm.Name == configMapName {
+
+		volumeName := "trusted-ca"
+		caVolumeMountPath := "/etc/pki/tls/certs/ca-bundle.crt"
+		odhVolumeMountPath := "/etc/pki/tls/certs/odh-ca-bundle.crt"
```

Comment:
`ca-bundle.crt`, exists in path `/etc/pki/tls/certs/` , this would override it.
it would be better if we add a prefix, like `custom-ca-bundle.crt`

please also update the Path in the volumes section.